### PR TITLE
Windows: Fix updating (#34)

### PR DESCRIPTION
**Pull request by Rummskartoffel**
- [x] Make installer try a few times
- [x] Make messenger properly invoke installer
- [x] Test

Extension-side changes are at tridactyl/tridactyl#3404.

### CODE REVIEW

**Comment by bovine3dom** [comment]
CI failure looks temporary. I'll try to remember to re-run it once more tomorrow.

File: src/native_main.nim [hunk 1]
```diff
@@ -137,6 +137,16 @@ proc handleMessage(msg: MessageRecv): MessageResp =
             result.code = some waitForExit(process)
             close(process)
 
+        of "run_async":
+            when defined(windows):
+                let command = "cmd /c " & msg.command.get()
+                createOrphanProcess(command)
+            else:
+                let command = msg.command.get()
+                discard startProcess(command, options = {poEvalCommand})
```
Comment (bovine3dom) [inline src/native_main.nim:146]: is this definitely `async` enough that it can be lives after native exits?

Comment (Rummskartoffel) [reply src/native_main.nim:146]: I'm not completely sure. [The doc](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging#closing_the_native_app) says you have to put the processes you want preserved into their own process group, but maybe the program I tested it with (`gnome_terminal`) does that anyway. I'll look into it more.

Comment (Rummskartoffel) [reply src/native_main.nim:146]: OK, I'm confused now. I took another look at `poDaemon` and it's *supposed to* make the process run in its own process group – which it does, but only when I use `gen_native_message.py` and not when I try it through Firefox. But either way, the process keeps running???

Comment (bovine3dom) [reply src/native_main.nim:146]: > and not when I try it through Firefox

I'd check that you're _definitely_ calling the version you think you're calling through Firefox before you spend too long debugging it.

Comment (Rummskartoffel) [reply src/native_main.nim:146]: No, I'm absolutely sure. I've now tried it on both WSL2 and in a Lubuntu VM, and it looks like `poDaemon` has no effect at all. When the native messenger is called manually, a new process group is created, and when it's called by Firefox, it isn't. Either way, the process doesn't get killed by Firefox.

Comment (bovine3dom) [reply src/native_main.nim:146]: In retrospect I think we already knew the process wouldn't be killed since `:restart` works easily on Linux.

**Comment by bovine3dom** [comment]
Any further issues with this?

**Comment by Rummskartoffel** [comment]
I still wanted to do a bit more testing, but I didn't have time to do it yet. Should be able to get it done tomorrow.

**Comment by Rummskartoffel** [comment]
OK, I think it's ready now. One issue I noticed (besides one I fixed over at tridactyl/tridactyl#3404) was that the old installer doesn't work on Windows 8, but that's none of this PR's business.

**Pull request merged**